### DISCUSSION TITLE
Keyboard shortcuts think I use QWERTY while I use AZERTY

### DISCUSSION THREAD

**Discussion by Haroenv** [community/community#5446] [Codespaces]
When I do command+Z (location of the w character in qwerty) to undo the window closes. I guess this might be happening if you listen to the key code instead of the character typed. 

I noticed layout: French was enabled though, so I'm not sure how that could be happening

In local vscode the right shortcuts are used.

I'm using Safari.

Not sure if there's anything else I can do to help you reproduce the issue?

**Comment by adityadees** [comment]
have you tried to rebuild your  [Command Palette](https://docs.github.com/en/codespaces/codespaces-reference/using-the-command-palette-in-codespaces) ? and [this](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette)

**Reply by Haroenv** [reply]
I guess it must have been some kind of a fluke, as when I try again now the keys get detected as normal. I noticed however that the keyboard layout said US until I started typing in the file itself (then it changed to French)
